libraries/js/text/regex.effekt

```diff
@@ -1,33 +1,242 @@
 module text/regex
 
 import string
+import option
+import list
+
 
 extern type Regex
 
-record Match(matched: String, index: Int)
+// matches and groups
+record Range(start: Int, end: Int)
+record Group(matched: String, index: Option[Range])
```

Comment:
Can we just always generate indices and turn this into `index: Range`?

Reply:
That's definitely possible and should decrease complexity